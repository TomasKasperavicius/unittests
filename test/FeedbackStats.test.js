import React from 'react';
import renderer from 'react-test-renderer';
import FeedbackStats from '../src/components/FeedbackStats';
import FeedbackContext from '../src/context/FeedbackContext';

describe('FeedbackStats component', () => {
  it('displays the number of reviews when there are no feedback', () => {
    const component = renderer.create(
      <FeedbackContext.Provider value={{ feedback: [] }}>
        <FeedbackStats />
      </FeedbackContext.Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children.join('')).toContain('0 Reviews');
  });

  it('displays the average rating', () => {
    const feedback = [
      { id: 1, rating: 4 },
      { id: 2, rating: 5 },
      { id: 3, rating: 3 },
      { id: 4, rating: 2 },
    ];

    const component = renderer.create(
      <FeedbackContext.Provider value={{ feedback }}>
        <FeedbackStats />
      </FeedbackContext.Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].children.join('')).toContain('Average Rating: 3.5');
  });

  it('displays a whole number for the average when all ratings are whole numbers', () => {
    const feedback = [
      { id: 1, rating: 3 },
      { id: 2, rating: 4 },
      { id: 3, rating: 2 },
    ];

    const component = renderer.create(
      <FeedbackContext.Provider value={{ feedback }}>
        <FeedbackStats />
      </FeedbackContext.Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].children.join('')).toContain('Average Rating: 3');
  });

  it('displays the correct number of reviews and average when there is only one feedback', () => {
    const feedback = [{ id: 1, rating: 5 }];

    const component = renderer.create(
      <FeedbackContext.Provider value={{ feedback }}>
        <FeedbackStats />
      </FeedbackContext.Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children.join('')).toContain('1 Review');
    expect(tree.children[1].children.join('')).toContain('Average Rating: 5');
  });
});
