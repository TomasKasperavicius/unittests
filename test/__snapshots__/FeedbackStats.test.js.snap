// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackStats component displays a whole number for the average when all ratings are whole numbers 1`] = `
<div
  className="feedback-stats"
>
  <h4>
    3
     Reviews
  </h4>
  <h4>
    Average Rating: 
    3
  </h4>
</div>
`;

exports[`FeedbackStats component displays the average rating 1`] = `
<div
  className="feedback-stats"
>
  <h4>
    4
     Reviews
  </h4>
  <h4>
    Average Rating: 
    3.5
  </h4>
</div>
`;

exports[`FeedbackStats component displays the correct number of reviews and average when there is only one feedback 1`] = `
<div
  className="feedback-stats"
>
  <h4>
    1
     Reviews
  </h4>
  <h4>
    Average Rating: 
    5
  </h4>
</div>
`;

exports[`FeedbackStats component displays the number of reviews when there are no feedback 1`] = `
<div
  className="feedback-stats"
>
  <h4>
    0
     Reviews
  </h4>
  <h4>
    Average Rating: 
    0
  </h4>
</div>
`;
